pkg/controller/controller.go

```diff
@@ -199,7 +200,18 @@ func (c *Controller) init() {
 }
 
 // Run is the main entrypoint for the controller.
-func (c *Controller) Run(stopCh <-chan struct{}) {
+func (c *Controller) Run(ctx context.Context) {
+	stopCh := make(chan struct{})
```

Comment:
Is it possible to keep the `stopCh` as a param?

Reply:
Could we keep the `stopCh` as a controller member, and return it when the controller is created, then we can just use it to stop here instead of requiring a context?